### ISSUE TITLE
StatefulSet Status is missing UpdatedReplicas field

### ISSUE THREAD

**Issue by andrewsykim** [kubernetes/kubernetes#52653]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT

/kind bug

**What happened**:
StatefulSet Status is missing `updatedReplicas` field but has `updateRevision` set
```
status:
  currentReplicas: 1
  currentRevision: consul-example-703895715
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updateRevision: consul-example-703895715
```

**What you expected to happen**:
According to the docs in https://kubernetes.io/docs/api-reference/v1.7/#statefulsetstatus-v1beta1-apps:
> updatedReplicas is the number of Pods created by the StatefulSet controller from the StatefulSet version indicated by updateRevision.

 It seems that `updatedReplicas` is actually only useful when doing a rolling update, if that is the case we should unset `updateRevision` to avoid confusion here. Otherwise, for a statefulset that is updated and not doing a rolling update, `updatedReplicas` should equal `currentReplicas`. 

**How to reproduce it (as minimally and precisely as possible)**:
Create a statefulset on v1.7.3

**Environment**:
- Kubernetes version (use `kubectl version`): v1.7.3

/sig apps

**Comment by faraazkhan** [comment]
This is actually causing `kubectl rollout status` to break for statefulsets, when they are created. 

Here is the behaviour, I have noticed:
1. `updatedReplicas` seems to be set correctly when updating an existing `statefulset`. 

2. But when creating a statefulset, `updatedReplicas` is not set. Causing this condition: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/rollout_status.go#L139 to always evaluate as true.  Which leads to `kubectl rollout status sts/<name>` to enter a never ending loop `Waiting for partitioned roll out to finish`.

3. This is different from how the deployment controller handles the `updatedReplicas` field, where it is simply the count of replicas in the new replica set:https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/sync.go#L493 and hence set correctly 

IMHO the fix here should be that we have consistent behavior for this field between `deployments` and `statefulsets`, which means we set `updatedReplicas` correctly when a `statefulset` is being created for the first time. 

Here is a simple way to replicate the issue:
Create a Statefulset using the manifest below:
```
cat <<EOF > nginx.yaml
---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: web
spec:
  serviceName: "nginx"
  replicas: 2
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: k8s.gcr.io/nginx-slim:0.8
        ports:
        - containerPort: 80
          name: web
        volumeMounts:
        - name: www
          mountPath: /usr/share/nginx/html
  volumeClaimTemplates:
  - metadata:
      name: www
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 1Gi
EOF
kubectl apply -f nginx-state.yaml && kubectl rollout status sts/web  
```
Notice how rollout status is stuck waiting for `updatedReplicas` to change and its never set. 
As shown here: https://www.screencast.com/t/6pcddPLKK

/remove-lifecycle stale

**Comment by andrewsykim** [comment]
Agreed, consistency with deployments would be nice

**Comment by jim-minter** [comment]
@mfojtik is this on your team's radar? `kubectl rollout status statefulset` hangs forever if a statefulset rollout is not actually happening at the time.

**Comment by linydquantil** [comment]
@jim-minter  same issue for me

**Comment by huyqut** [comment]
Are there any updates on this problem?

**Comment by sfudeus** [comment]
@huyqut To my understanding, this is fixed in master with PR #62943, but waiting for cherrypicking to 1.11 in #67570
